crates/zed/src/languages/terraform.rs

```diff
@@ -0,0 +1,184 @@
+use async_trait::async_trait;
+use collections::HashMap;
+use futures::StreamExt;
+pub use language::*;
+use lsp::{CodeActionKind, LanguageServerBinary};
+use smol::fs::{self, File};
+use std::{any::Any, ffi::OsString, path::PathBuf, str};
+use util::{
+    async_maybe,
+    fs::remove_matching,
+    github::{latest_github_release, GitHubLspBinaryVersion},
+    ResultExt,
+};
+
+fn terraform_ls_binary_arguments() -> Vec<OsString> {
+    vec!["serve".into()]
+}
+
+pub struct TerraformLspAdapter;
+
+#[async_trait]
+impl LspAdapter for TerraformLspAdapter {
+    fn name(&self) -> LanguageServerName {
+        LanguageServerName("terraform-ls".into())
+    }
+
+    fn short_name(&self) -> &'static str {
+        "terraform-ls"
+    }
+
+    async fn fetch_latest_server_version(
+        &self,
+        delegate: &dyn LspAdapterDelegate,
+    ) -> Result<Box<dyn 'static + Send + Any>> {
+        // TODO: maybe use release API instead
+        // https://api.releases.hashicorp.com/v1/releases/terraform-ls?limit=1
+        let release = latest_github_release(
+            "hashicorp/terraform-ls",
+            false,
+            false,
+            delegate.http_client(),
+        )
+        .await?;
+
+        Ok(Box::new(GitHubLspBinaryVersion {
+            name: release.tag_name,
+            url: Default::default(),
+        }))
+    }
+
+    async fn fetch_server_binary(
+        &self,
+        version: Box<dyn 'static + Send + Any>,
+        container_dir: PathBuf,
+        delegate: &dyn LspAdapterDelegate,
+    ) -> Result<LanguageServerBinary> {
+        let version = version.downcast::<GitHubLspBinaryVersion>().unwrap();
+        let zip_path = container_dir.join(format!("terraform-ls_{}.zip", version.name));
+        let version_dir = container_dir.join(format!("terraform-ls_{}", version.name));
+        let binary_path = version_dir.join("terraform-ls");
+        let url = build_download_url(version.name)?;
+
+        if fs::metadata(&binary_path).await.is_err() {
+            let mut response = delegate
+                .http_client()
+                .get(&url, Default::default(), true)
+                .await
+                .context("error downloading release")?;
+            let mut file = File::create(&zip_path).await?;
+            if !response.status().is_success() {
+                Err(anyhow!(
+                    "download failed with status {}",
+                    response.status().to_string()
+                ))?;
+            }
+            futures::io::copy(response.body_mut(), &mut file).await?;
+
+            let unzip_status = smol::process::Command::new("unzip")
+                .current_dir(&container_dir)
+                .arg(&zip_path)
+                .arg("-d")
+                .arg(&version_dir)
+                .output()
+                .await?
+                .status;
+            if !unzip_status.success() {
+                Err(anyhow!("failed to unzip deno archive"))?;
```

Comment:
This seems to not been updated ☺️ 
```suggestion
                Err(anyhow!("failed to unzip terraform archive"))?;
```

Reply:
Good catch!